tlsredis.go

```diff
@@ -137,17 +135,14 @@ func GetClient(opts *Options) (*redis.Client, error) {
 		}
 	}
 
-	opt := redis.Options{
-		Dialer:   dialFunc,
-		DB:       db,
-		PoolSize: opts.PoolSize,
-	}
+	opts.Dialer = dialFunc
+	opts.DB = db
 	if u.User != nil {
 		redisPass, _ := u.User.Password()
-		opt.Password = redisPass
+		opts.Password = redisPass
 	}
 
-	rc := redis.NewClient(&opt)
+	rc := redis.NewClient(&opts.Options)
```

Comment:
Options are passed by reference and `redis.NewClient` does indeed seem to make use of this privilege with gusto, so I think the previous scheme where a newly allocated `redis.Options` gets passed in each call to `redis.NewClient` is safer.